NServiceBus.FileBasedRouting/XmlRoutingFile.cs

```diff
@@ -2,42 +2,83 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Reflection;
+using System.Xml;
 using System.Xml.Linq;
+using System.Xml.Schema;
 
 namespace NServiceBus.FileBasedRouting
 {
-    class XmlRoutingFile
+    public class XmlRoutingFile
     {
-        private readonly string filePath;
-
-        public XmlRoutingFile(string filePath)
+        public XmlRoutingFile(Func<Stream> streamAccessor)
```

Comment:
would it make more sense to rename this class to `XmlRoutingFileParser` and just pass a XDocument to it?